### DISCUSSION TITLE
Question with regard to `AsyncEventingBasicConsumer`

### DISCUSSION THREAD

**Discussion by EmanueleBaron** [rabbitmq/rabbitmq-dotnet-client#1565] [Q&A] [answered]
My code:

```
using var channel = connection.CreateModel());
var consumer = new AsyncEventingBasicConsumer(channel);
consumer.Received += async (co, ea) =>
{
    await Task.Delay(100);
};
channel.BasicConsume(queue, false, consumer);
```

I received this error: `Exceção gerada: 'System.ObjectDisposedException' em RabbitMQ.Client.dll`

**Comment by lukebakken** [comment]
Hello, thanks for using this library.

I converted your issue to a discussion since there is no evidence of a bug in this library.

When reporting issues with this library, or any software in general, it's best to provide enough information for someone to reproduce the problem. In your case, that involves runnable code.

Since you didn't provide that, I created the following git repository with enough code to be runnable:

https://github.com/lukebakken/rabbitmq-dotnet-client-1564/blob/main/Program.cs

When I run the program, it consumes messages from an exclusive queue. I can use the Management UI to publish a message to that queue, and everything works as expected:

```
C:\Users\lbakken\development\lukebakken\rabbitmq-dotnet-client-1564 [main ≡]
> dotnet run
[INFO] hit [Enter] to exit
[INFO] message body: Hello message
```

If I'm not using the correct code, please fork my repository and make the necessary changes to reproduce the issue.

**Comment by EmanueleBaron** [comment]
The your code worked, but when uncomment the line Console.ReadLine();, the messages in the queue are not process, why?
In my method, I'm not have a 'Console.ReadLine();' why is a worker aplicattion.
![Captura de tela 2024-05-13 155843](https://github.com/rabbitmq/rabbitmq-dotnet-client/assets/14804392/58febac5-5303-48c9-8625-7557666bcbfe)

**Reply by lukebakken** [reply]
> The your code worked, but when uncomment the line Console.ReadLine();, the messages in the queue are not process, why?

Nothing is keeping the main thread running, and the program exits well before the data is processed. This is how this library has always worked.

If you can provide your worker application code, I can provide suggestions.

**Reply by EmanueleBaron** [reply]
Right, I created a fork your repository with my worker application : https://github.com/EmanueleBaron/rabbitmq-dotnet-client-1564.git
In my code I receveid this error: Object name: 'RabbitMQ.Client.Impl.AutorecoveringModel'.
![Captura de tela 2024-05-13 165911](https://github.com/rabbitmq/rabbitmq-dotnet-client/assets/14804392/7f39112b-0570-472c-b98b-91c5333d589c)
Thanks your support

**Reply by lukebakken** [reply, marked as answer]
I can't push to your repository, because you did not open a pull request (https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests)

However, you can see my changes here:

https://github.com/lukebakken/rabbitmq-dotnet-client-1564/tree/gh-1/main

If you add my repository as another remote to your local clone, you can pull those changes into your code. Something like this:

```
cd path/to/rabbitmq-dotnet-client-1564
git remote add lukebakken https://github.com/lukebakken/rabbitmq-dotnet-client-1564.git
git fetch lukebakken
git checkout main
git pull lukebakken/gh-1/main
```

**Reply by EmanueleBaron** [reply]
Thank you very much!!! now it's working. :D

I'm sorry this repository...
